source/detail/parser/x86_64.cpp

```diff
@@ -0,0 +1,330 @@
+// Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+// Spack Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#include <iostream>
```

Comment:
This is a bit of a nit - but I'm a stickler for making sure that files are renamed (and not re-created) in the git history so the author is maintained. I was the primary author of this content and it took me a long time (I'm sure much longer than it would have taken you!) But if it's okay with you, I'd like to PR this to your branch so that I'm given credit.

Reply:
I've copied the file locally, so if you want to delete it from your branch I'll have a PR with it asap.

Reply:
This wasn't a rename because I extracted only a subset of the code from corpora.cpp into here. I agree that maintaining provenance is important, but git doesn't have a good way to tracking that kind of refactoring.

Reply:
Understood! But I think if you had moved it without change, and then commit, and then made changes, the file would be shown as a rename (with some changes) instead of an entirely new file (with previous git history lost).